db/db_test.go

```diff
@@ -91,6 +92,36 @@ func TestE2EWithThreads(t *testing.T) {
 	}
 }
 
+func TestMissingCollection(t *testing.T) {
+	t.Parallel()
+
+	tmpDir, err := ioutil.TempDir("", "")
+	checkErr(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	n, err := common.DefaultNetwork(tmpDir, common.WithNetDebug(true), common.WithNetHostAddr(util.FreeLocalAddr()))
+	checkErr(t, err)
+	defer n.Close()
+
+	id := thread.NewIDV1(thread.Raw, 32)
+	db, err := NewDB(context.Background(), n, id, WithNewRepoPath(tmpDir))
+	checkErr(t, err)
+	defer db.Close()
+	c, err := db.NewCollection(CollectionConfig{
+		Name:   "dummy",
+		Schema: util.SchemaFromInstance(&dummy{}, false),
+	})
+	checkErr(t, err)
+	// Delete collection from map to "simulate" not having received it yet...
+	// Motivated by https://github.com/textileio/go-threads/issues/379
+	delete(db.collections, "dummy")
```

Comment:
Kinda hacky, but a simple simulation of a pair of peers where one doesn't know about the collections created on the other. This highlights the need to think again about how we might handle this more robustly in the future. For now, its useful to have this test.